### DISCUSSION TITLE
Clone instructions do not mention submodule

### DISCUSSION THREAD

**Discussion by VictorEijkhout** [idaholab/moose#26978] [Q&A Getting Started]
You're pulling in a lot of submodules, but the instructions under "Cloning MOOSE" do not mention that. As far as I can tell.

**Comment by milljm** [comment]
The reason we don't go to mention submodules during our "Getting Started" instructions, is that the user is either going to use Conda, which will include pre-built packages of all the submodules; thus not requiring any interaction with submodule what so ever.

Or they are going to build the dependencies themselves using the scripts found in `moose/scripts`. These scripts perform all the `git submodule` magic necessary.

I think the only interaction a user will require to understand the intricacies involved with submodules, is when they are actively committing to MOOSE; asking for a newer version of PETSc, for example.

I was going to convert this over to a discussion, but I am willing to leave it open as an issue, if we find that we should add submodule documentation that may be lacking elsewhere. But I truly believe it does not belong in the Getting Starting instructions...

**Comment by GiudGiud** [comment]
Converting to discussions, this is not an issue. 

We do not pull in any submodule by default. Only developers pull in libmesh & petsc, and some advanced users of the modules will use the saline, neml2, thermochimica etc submodules.

**Comment by VictorEijkhout** [comment]
Ok, you can close this. Please address the other discussion that I'm failing to build from source.
